CMakeLists.txt

```diff
@@ -260,15 +274,19 @@ else()
 endif()
 
 # target_compile_options was added in CMake 2.8.12 and is the recommended way to
-# set compile flags. Note that this sets the COMPILE_OPTIONS property (also
-# available only in 2.8.12+) rather than the COMPILE_FLAGS property, which is
-# deprecated. The former can handle lists whereas the latter cannot.
-if(CMAKE_VERSION VERSION_LESS 4.8.12)
-  string(REPLACE ";" " " f90flags "${f90flags}")
-  set_property(TARGET ${program} PROPERTY COMPILE_FLAGS "${f90flags}")
-else()
-  target_compile_options(${program} PUBLIC ${f90flags})
-endif()
+# set compile flag, but it doesn't seem to work with both Fortran and C.  In
+# theory the commented code below might work but I couldn't get it running.
+# Maybe it requires a newer version of CMake?
+#target_compile_options(${program} PUBLIC 
+#  $<$<COMPILE_LANGUAGE:Fortran>:${f90flags}>
+#  $<$<COMPILE_LANGUAGE:C>:${cflags}>)
+# This is not the recommended method, but it is a functional method.  Convert
+# the semicolon-delineated lists into space-delineated ones then set them as the
+# compiler flags.
```

Comment:
Perhaps an easier way to handle this is to just build Faddeeva.cc as a library and then link OpenMC against that library. That way, since it has its own target, you can set compile flags for it separately.